### PR TITLE
538

### LINKED ISSUE #538: Add Resync Button
## User Story
As a user, if I am a vendor I want an indicator that I may need to resync my wallet when necessary, and a way to easily do so.

### Transactions
- [ ] In the transactions screen, in the sales tab, if the last transaction is more than 24 hours old and is still unfunded, show a message with a link that will open the Settings on the Advanced Tab.
- [ ] The message: "You may need to resync your wallet to receive payments for one or more unfunded orders. [link]Open Settings and click the Resync button.[/link]"

### Settings
- [x] In Settings/Advanced, there should be a Resync button.
- [x] Clicking the button will call (POST) the /wallet/resyncblockchain API with no parameters.
- [ ] Clicking the button will save a timestamp in local storage.
- [ ] If the saved timestamp is less than an hour old, the Resync button should be disabled.
- [x] The Resync button should have text next to it "Resyncing the wallet can take several minutes."
- [ ] After it is clicked, the Resync button should have text near it "Check your transactions for updates in a few minutes. You can only resync once an hour."

### PR DESCRIPTION
**Pull request by rmisio**
closes #538

### CODE REVIEW

File: js/utils/resyncBlockchain.js [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+  ...Events,
+};
+
+export { events };
+
+let server = getServer();
+
+// If you change this, be sure to change anywhere in the GUI you may have output how
+// long its unavailable (e.g. Advanced Settings).
+const resyncInactiveTime = 1000 * 60 * 60 * 1; // 1 hour
```
Comment (rmisio) [inline js/utils/resyncBlockchain.js]: You could muck with this if you want to test a resync being unavailable for a certain amount of time and not having to wait the hour that's coded here.

File: js/views/transactions/SalesTab.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+      if (this.resyncPopinMessage) {
+        this.resyncPopinMessage.setState({
+          isSyncing: false,
+        });
+      }
+    });
+  }
+
+  get transactionAgeForResync() {
+    return 1000 * 60 * 60 * 24; // 24 hours
```
Comment (rmisio) [inline js/views/transactions/SalesTab.js]: You could muck with this if you want to see the resync pop-up on the sales tab and not have to wait for a transaction to be unconfirmed for the 24 hours listed here.

File: js/views/modals/Settings/advanced/ReloadTransactions.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    };
+
+    super(opts);
+    this.listenTo(resyncEvents, 'resyncing',
+      () => this.setState({
+        isSyncing: true,
+        syncComplete: false,
+      }));
+    this.listenTo(resyncEvents, 'resyncComplete',
+      () => this.setState({ isSyncing: false }));
```
Comment (jjeffryes) [inline js/views/modals/Settings/advanced/ReloadTransactions.js]: This needs syncComplete: true, or the button in settings will never show the complete state.

File: js/views/transactions/SalesTab.js [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+    const linkText = app.polyglot.t('transactions.sales.resyncPopInMessage.link');
+    const needSyncText = app.polyglot.t('transactions.sales.resyncPopInMessage.message', {
+      icon: '<span class="ion-alert-circled"></span>',
+      link: `<a class="clrTEm js-resync">${linkText}</a>`,
+      text: `<b>${app.polyglot.t('transactions.sales.resyncPopInMessage.text')}</a>`,
+    });
+
+    this.resyncPopinMessage = this.createChild(ResyncPopInMessage, {
+      messageText: '-----', // not used but needed so the pop in message base class dosn't cry
```
Comment (jjeffryes) [inline js/views/transactions/SalesTab.js]: "doesn't"

File: js/templates/transactions/resyncPopInMessage.html [hunk 5]
```diff
@@ -0,0 +1,23 @@
+<% if (ob.dismissable) { %>
+<a class="closeIcon tx2 js-dismiss">
+  <span class="ion-ios-close-empty clrBr clrP clrT"></span>
```
[the remark this answers is not in the document]
Comment (morebrownies) [reply js/templates/transactions/resyncPopInMessage.html]: Dark border on the close button sounds good to me too.

**Pull request merged**